src/App.tsx

```diff
@@ -1,24 +1,200 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { NewTodo, Todo } from './types/Todo';
+import { Status } from './types/Statuses';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Errors } from './types/Errors';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoItem } from './components/TodoItem';
+import { USER_ID } from './helpers/userId';
+import {
+  addTodo,
+  updateTodo,
+  deleteTodo,
+  getTodos,
+} from './api/todos';
 
-const USER_ID = 0;
+const applyFilter = (todos: Todo[], filter: Status) => {
+  const conditions = {
+    [Status.All]: () => {
+      return todos;
+    },
+    [Status.Active]: () => {
+      return todos.filter(todo => !todo.completed);
+    },
+    [Status.Completed]: () => {
+      return todos.filter(todo => todo.completed);
+    },
+  };
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  if (filter in conditions) {
+    return conditions[filter]();
   }
 
+  return todos;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState<Errors | ''>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+  const [title, setTitle] = useState('');
+  const [currentId, setCurrentId] = useState<number | null>(null);
+  const [submittedTitle, setSubmittedTitle] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage(Errors.LoadError));
+  }, []);
+
+  const filteredTodos = applyFilter(todos, filter);
+
+  const handleUpdateTodo = async (updatedTodo: Todo): Promise<void> => {
+    try {
+      const result = await updateTodo(updatedTodo);
+
+      setTodos((currentTodos) => {
+        const newTodos = currentTodos.map((item) => (
+          item.id === result.id ? result : item
+        ));
+
+        return newTodos;
+      });
+
+      setTempTodo((prevTempTodo) => {
+        if (!prevTempTodo) {
+          return null;
+        }
+
+        return {
+          ...prevTempTodo,
+          title: result.title,
+          id: result.id,
+          userId: result.userId,
+          completed: result.completed,
+        };
+      });
```

Comment:
why do you need to work with tempTodo during updating? we use tempTodo only for adding a new one